src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -315,6 +319,21 @@ private void EmitExpressionCore(BoundExpression expression, bool used)
             }
         }
 
+        private void EmitThrowExpression(BoundThrowExpression node, bool used)
+        {
+            this.EmitExpression(node.Expression, true);
+            var thrownType = node.Expression.Type;
+            if (thrownType?.TypeKind == TypeKind.TypeParameter)
```

Comment:
This is precisely the same way the throw statement is handled. I will make them both defer to the same helper.

---
In reply to: [76669674](https://github.com/dotnet/roslyn/pull/13437#discussion_r76669674) [](ancestors = 76669674)